NBlockchain/Services/Net/TcpPeerNetwork.cs

```diff
@@ -45,16 +46,17 @@ public class TcpPeerNetwork : IPeerNetwork, IDisposable
         private Timer _sharePeersTimer;
         private Timer _discoveryTimer;
 
-        private string _internalConnsctionString;
-        private string _externalConnsctionString;
+        private string _internalConnectionString;
+        private string _externalConnectionString;
 
         private object _duplicateLock = new object();
 
         public Guid NodeId { get; private set; }
 
-        public TcpPeerNetwork(uint port, IBlockRepository blockRepository, IEnumerable<IPeerDiscoveryService> discoveryServices, ILoggerFactory loggerFactory, IOwnAddressResolver ownAddressResolver, IUnconfirmedTransactionPool unconfirmedTransactionPool, IReceiver reciever)
+        public TcpPeerNetwork(uint port, INatTraversal natTraversal, IBlockRepository blockRepository, IEnumerable<IPeerDiscoveryService> discoveryServices, ILoggerFactory loggerFactory, IOwnAddressResolver ownAddressResolver, IUnconfirmedTransactionPool unconfirmedTransactionPool, IReceiver reciever)
```

Comment:
Should we maybe take an `IEnumerable<INatTraversal>` and then iterate through the list until the first successful one?